web/assets/js/GoogleAnalyticsService.js

```diff
@@ -0,0 +1,76 @@
+class GoogleAnalyticsService {
+
+    constructor () {
+        this.gaDataAttr = 'data-ga-action';
+        this.gaCategoryAttr = 'data-ga-category';
+        this.gaLabelAttr = 'data-ga-label';
+    }
+
+    rewriteByChildSelector (selector) {
+        if (selector.classList.contains('ga-link-listener')) {
+            const childSelector = selector.querySelector('a');
+
+            return this.rewriteDataAttributes(selector, childSelector);
+        }
+
+        return selector;
+    }
+
+    getGAItemObject (selector) {
+        return new GoogleAnalyticsItem(
+            'event',
+            this.getGACategory(),
+            this.getGAAction(selector),
+            this.getGALabel(selector)
+        );
+    }
+
+    getGAAction (selector) {
+        return selector.getAttribute('data-ga-action');
+    }
+
+    getGACategory () {
+        const body = document.body;
+
+        return body.hasAttribute(this.gaCategoryAttr) ? Utils.capitalize(body.getAttribute(this.gaCategoryAttr)) : null;
+    }
+
+    getGALabel (selector) {
+        if (typeof selector !== 'undefined') {
+            return selector.hasAttribute(this.gaLabelAttr) ? selector.getAttribute(this.gaLabelAttr) : null;
+        }
+    }
+
+    sendGARequest (gaItem) {
+        if (gaItem instanceof GoogleAnalyticsItem) {
+            ga(
+                'send',
+                gaItem.hitType,
+                gaItem.eventCategory,
+                gaItem.eventAction,
+                gaItem.eventLabel = typeof gaItem.eventLabel !== 'undefined' ? gaItem.eventLabel : delete gaItem.eventLabel,
+                gaItem.transport = typeof gaItem.transport !== 'undefined' ? gaItem.transport : delete gaItem.transport,
```

Comment:
`typeof gaItem.transport` is never `undefined` as it has the default value.

`delete gaItem.transport`? I think `delete` work for objects.

Even more, I don't get the idea of this line. If something is undefined, delete the undefined?

Reply:
Ok, my mistake. I'm not an expert in javascript. This is why i asked Piotr for review. This is very helpful for me. I thought that in this way I can secure passing data to be sure that always we will send properly request without unnecessary data. I don't know why I thought in javascript even when you have default value for property and when you instantiate object and you not pass value for one of property then it will be undefined. I will change this.